server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -127,6 +130,33 @@ public Task register(String type, String action, TaskAwareRequest request) {
         return task;
     }
 
+    public <Request extends ActionRequest, Response extends ActionResponse>
+    Task registerAndExecute(String type, TransportAction<Request, Response> action, Request request,
```

Comment:
I agree that placement of this method in TransportAction might not be ideal, but I feel like TaskManager shouldn't be in the business of executing actions, it is busy enough registering and unregistering things. Would it make more sense to just place this code into NodeClient or TransportService?

Reply:
I'm open to either of those places, but a quick try at moving it to either resulted in awkwardness, particularly in tests needing to initialize more (both of those classes are more heavyweight to construct for tests). Most likely the TransportService will make the most sense in the long run since I want to replace/move the actionfilter chain there, but I would like to wait until it is more clear how that will look before settling on a particular location.